### PR TITLE
jasmine: add config interfaces

### PR DESCRIPTION
**Pull request by UziTech**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://jasmine.github.io/api/npm/3.8/Jasmine>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -197,8 +197,9 @@ declare function spyOnProperty<T>(object: T, property: keyof T, accessType?: "ge
 /**
  * Installs spies on all writable and configurable properties of an object.
  * @param object The object upon which to install the `Spy`s.
+ * @param includeNonEnumerable Whether or not to add spies to non-enumerable properties.
  */
-declare function spyOnAllFunctions<T>(object: T): jasmine.SpyObj<T>;
+declare function spyOnAllFunctions<T>(object: T, includeNonEnumerable?: boolean): jasmine.SpyObj<T>;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:202]: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54255 covers this, so just be sure that your rebase handles this right.

File: types/jasmine/index.d.ts [hunk 2]
```diff
@@ -1182,30 +1252,58 @@ declare module "jasmine" {
...
+         */
         addSpecFile(filePath: string): void;
         addSpecFiles(files: string[]): void;
         addHelperFiles(files: string[]): void;
         addRequires(files: string[]): void;
-        configureDefaultReporter(options: any, ...args: any[]): void;
+        /**
+         * Configure the default reporter.
+         */
+        configureDefaultReporter(options: jasmine.DefaultReporterOptions): void;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:1271]: I can't find this in https://jasmine.github.io/api/npm/edge/Jasmine.html, could you point me to it?

Comment (UziTech) [reply types/jasmine/index.d.ts:1271]: It is not on the documentation site as a public property but it was already in the types so I figured it should probably be updated to be correct.

Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1271]: so it's a bogus declaration, should not be there IMO:
https://github.com/jasmine/jasmine/issues/1266
(that is the closes thing I can find)

Comment (chivesrs) [reply types/jasmine/index.d.ts:1271]: OK, here it is https://jasmine.github.io/setup/nodejs.html#reporters

File: types/jasmine/index.d.ts [hunk 3]
```diff
@@ -1164,6 +1165,75 @@ declare namespace jasmine {
...
+        showColors?: boolean;
+        jasmineCorePath?: string;
+        stackFilter?: any;
+        /**
+         * A function that takes a random seed and returns the command to reproduce
+         * that seed. Use this to customize the output when using ConsoleReporter
+         * in a different command line tool.
+         */
+        randomSeedReproductionCmd?: any;
+        order?: Order;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:1235]: stackFilter, random... order, I cannot find in the source code
https://github.com/jasmine/jasmine-npm/blob/main/lib/jasmine.js#L143

The first 4 I could. Should you remove the last 3?

File: types/jasmine/index.d.ts (hunk 2 above)
Comment (chivesrs) [reply types/jasmine/index.d.ts:1271]: Jasmine NPM is a weird case where you have to dig into the source code, Its documentation is kinda lacking, or if it's there, it's in the NodeJS specific file and not with everything else.

File: types/jasmine/index.d.ts (hunk 3 above)
Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1235]: cool, that's inline instantiated func, here is the soure:
```js
jasmine.configureDefaultReporter.toString()
'function(options) {\n' +
  '  options.timer = options.timer || new this.jasmine.Timer();\n' +
  '  options.print = options.print || function() {\n' +
  '    process.stdout.write(util.format.apply(this, arguments));\n' +
  '  };\n' +
  "  options.showColors = options.hasOwnProperty('showColors') ? options.showColors : true;\n" +
  '  options.jasmineCorePath = options.jasmineCorePath || this.jasmineCorePath;\n' +
  '\n' +
  '  this.reporter.setOptions(options);\n' +
  '  this.defaultReporterConfigured = true;\n' +
  '}'
  ```

Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1235]: So yes, only 4 top options, please.

File: types/jasmine/index.d.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1271]: agree, found this, updated your comment on options

File: types/jasmine/index.d.ts [hunk 4]
```diff
@@ -1182,30 +1252,58 @@ declare module "jasmine" {
...
+         * Provide a fallback reporter if no other reporters have been specified.
+         */
         provideFallbackReporter(reporter: jasmine.CustomReporter): void;
+        /**
+         * Clears all registered reporters.
+         */
         clearReporters(): void;
-        randomizeTests(value?: boolean): void;
+        /**
+         * Sets whether to randomize the order of specs.
```
Comment (peterblazejewicz) [inline types/jasmine/index.d.ts:1289]:
```suggestion
         * Sets whether to randomize the order of specs.
         * @deprecated	    
```
https://github.com/jasmine/jasmine/blob/163f93d6ffe905a7d2f907f83cea934706b69e64/src/core/Env.js#L541

File: types/jasmine/index.d.ts [hunk 5]
```diff
@@ -1164,6 +1165,75 @@ declare namespace jasmine {
...
+    interface JasmineOptions {
+        /**
+         * The path to the project's base directory. This can be absolute or relative
+         * to the current working directory. If it isn't specified, the current working
+         * directory will be used.
+         */
+        projectBaseDir?: string;
+    }
+
+    interface JasmineConfig {
```
Comment (peterblazejewicz) [inline types/jasmine/index.d.ts:1178]: Where is `seed` prop?
https://github.com/jasmine/jasmine/blob/163f93d6ffe905a7d2f907f83cea934706b69e64/src/core/Env.js#L61

Comment (UziTech) [reply types/jasmine/index.d.ts:1178]: This interface is for jasmine-npm `loadConfig` method not `env.configure`

[docs](https://jasmine.github.io/api/npm/edge/Configuration.html)

File: types/jasmine/index.d.ts (hunk 4 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:1281]: This function isn't deprecated https://github.com/jasmine/jasmine-npm/blob/main/lib/jasmine.js#L70-L78 only `jasmine.env.randomizeTests()` is

File: types/jasmine/index.d.ts (hunk 5 above)
Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1178]: Really sry 🤕

File: types/jasmine/index.d.ts (hunk 4 above)
Comment (peterblazejewicz) [reply types/jasmine/index.d.ts:1281]: sry, same topic (I've looked into wrong place, should catch up with optitian r somethng)

**Review by chivesrs — APPROVED** [review]

File: types/jasmine/index.d.ts (hunk 2 above)
Comment (sgravrock) [reply types/jasmine/index.d.ts:1263]: Yeah, the API docs for -npm literally didn't exist a week ago. What's there now is just a first pass. It's pretty likely that there are some things missing from the docs that ought to be public.

**Comment by elibarzilay** [comment]
(@UziTech, just to clarify, I made it available to merge if all of the discussions are resolved...)

**Review by peterblazejewicz — APPROVED** [review]

**Comment by UziTech** [comment]
Ready to merge

**Pull request merged**